docs/real-time-analytics/location-based-alerting/process.sql

```diff
@@ -0,0 +1,84 @@
+SET 'auto.offset.reset' = 'earliest';
+
+-- Table of merchant data including the calculated geohash
+CREATE TABLE merchant_locations (
+  id INT PRIMARY KEY,
+  description VARCHAR,
+  latitude DECIMAL(10,7),
+  longitude DECIMAL(10,7),
+  geohash VARCHAR
+) WITH (
+  KAFKA_TOPIC='merchant-locations', 
+  VALUE_FORMAT='JSON',
+  PARTITIONS = 6
+);
+
+-- A table to lookup merchants based on the a substring (precision) of the geohash
+CREATE TABLE merchants_by_geohash
+WITH (
+  KAFKA_TOPIC='merchant-geohash', 
+  FORMAT='JSON', 
+  PARTITIONS=6
+) AS 
+SELECT 
+  SUBSTRING(geohash, 1, 6) AS geohash, 
+  COLLECT_LIST(id) as id_list
+FROM merchant_locations
+GROUP BY SUBSTRING(geohash, 1, 6);
+
+-- Stream of user location data including the calculated geohash
+CREATE STREAM user_locations (
+  id INT,
+  latitude DECIMAL(10,7),
+  longitude DECIMAL(10,7),
+  geohash VARCHAR
+) WITH (
+  KAFKA_TOPIC='user-locations', 
+  VALUE_FORMAT='JSON',
+  PARTITIONS=6
+);
+
+-- Stream of alerts when a user's geohash based location roughly 
+-- intersects a collection of merchants locations
+CREATE STREAM alerts_raw
+WITH (
+  KAFKA_TOPIC='alerts-raw', 
+  VALUE_FORMAT='JSON', 
+  PARTITIONS=6
+) AS 
+SELECT 
+  user_locations.id as user_id,
+  user_locations.latitude AS user_latitude,
+  user_locations.longitude AS user_longitude,
+  SUBSTRING(user_locations.geohash, 1, 6) AS user_geohash,
+  EXPLODE(merchants_by_geohash.id_list) AS merchant_id
+FROM user_locations
+LEFT JOIN merchants_by_geohash ON SUBSTRING(user_locations.geohash, 1, 6) = 
+  merchants_by_geohash.geohash
+PARTITION BY null;
```

Comment:
Why do we need this clause?

Reply:
I wasn't sure either, the examples were based off the associated demo in the GH issue, here: https://github.com/confluentinc/ksqldb-recipes/issues/114

Should this query work without specifying the `PARTITION BY` ?

Reply:
I believe yes (but there could also be something about ksqlDB that I don't know of...) -- Mind trying it out and let me know?

Reply:
@rspurgeon can you review this before merge?

Reply:
Removing `PARTITION BY null` results in:
```
Could not determine output schema for query due to error: Key missing from projection. The query used to build `ALERTS_RAW` must include the join expression MERCHANTS_BY_GEOHASH.GEOHASH in its projection. Statement: CREATE STREAM ALERTS_RAW WITH (KAFKA_TOPIC='alerts-raw', PARTITIONS=6, REPLICAS=3, VALUE_FORMAT='JSON') AS SELECT USER_LOCATIONS.ID USER_ID, USER_LOCATIONS.LATITUDE USER_LATITUDE, USER_LOCATIONS.LONGITUDE USER_LONGITUDE, SUBSTRING(USER_LOCATIONS.GEOHASH, 1, 6) USER_GEOHASH, EXPLODE(MERCHANTS_BY_GEOHASH.ID_LIST) MERCHANT_ID FROM USER_LOCATIONS USER_LOCATIONS LEFT OUTER JOIN MERCHANTS_BY_GEOHASH MERCHANTS_BY_GEOHASH ON ((SUBSTRING(USER_LOCATIONS.GEOHASH, 1, 6) = MERCHANTS_BY_GEOHASH.GEOHASH)) EMIT CHANGES;
```

Reply:
Thanks. I understand.

It's something we need to fix on our side... It's an unnecessary required that a result STREAM needs a key.